### PR TITLE
Tracing ark0.4

### PR DESCRIPTION
**Pull request by findora-sun**
* **The major changes of this PR**

* **The major impacts of this PR**
  - [ ] Impact WASM?
  - [ ] Impact mainnet data compatibility?

* **Extra documentations**

### CODE REVIEW

File: algebra/src/ed25519/g1.rs [hunk 1]
```diff
@@ -196,31 +196,31 @@ impl Ed25519Point {
...
     pub fn get_y(&self) -> ZorroScalar {
-        let affine_point = G1Affine::from(self.0);
+        let affine_point = EdwardsAffine::from(self.0);
         ZorroScalar(affine_point.y)
     }
 
     /// Obtain a point using the x coordinate (which would be ZorroScalar).
-    pub fn get_point_from_x(x: &ZorroScalar) -> Result<Self> {
-        let point = G1Affine::get_point_from_x_old(x.0.clone(), false)
+    pub fn get_point_from_y(y: &ZorroScalar) -> Result<Self> {
```
Comment (weikengchen) [inline algebra/src/ed25519/g1.rs:210]: Is it true that we never use this function yet?

File: algebra/src/ed25519/g1.rs [hunk 2]
```diff
@@ -196,31 +196,31 @@ impl Ed25519Point {
...
     }
     /// Get the y-coordinate of the ed25519 affine point.
     #[inline]
     pub fn get_y(&self) -> ZorroScalar {
-        let affine_point = G1Affine::from(self.0);
+        let affine_point = EdwardsAffine::from(self.0);
         ZorroScalar(affine_point.y)
     }
 
     /// Obtain a point using the x coordinate (which would be ZorroScalar).
```
Comment (weikengchen) [inline algebra/src/ed25519/g1.rs:209]:
```suggestion
    /// Obtain a point using the y coordinate (which would be ZorroScalar).
```

File: algebra/src/ed25519/g1.rs (hunk 1 above)
Comment (findora-sun) [reply algebra/src/ed25519/g1.rs:210]: Yeah.

File: api/src/gen-params.rs [hunk 3]
```diff
@@ -55,6 +58,9 @@ enum Actions {
     /// Generates the uniform reference string for Bulletproof (over the Zorro curve),
     BULLETPROOF_OVER_ZORRO { directory: PathBuf },
 
+    /// Cut the SRS, adapt to lagrange calculate, and only save the minimum 2^11, and 2^12, 2^13 padding
```
Comment (weikengchen) [inline api/src/gen-params.rs:61]:
```suggestion
    /// Cut the SRS, adapt to Lagrange, and only save the minimum 2^11, 2^12, and 2^13 padding
```

**Review by weikengchen — APPROVED** [review]

**Pull request merged**